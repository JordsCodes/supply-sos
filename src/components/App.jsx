import React, { useState } from "react";
import { createBrowserRouter, RouterProvider } from "react-router-dom";
import Root from "./Root";
import LandingPage from "./LandingPage";
import YearGroup from "./YearGroup";
import Subject from "./Subject";
import Topic from "./Topic";
import CurriculumAim from "./CurriculumAim";
import LessonPlan from "./LessonPlan";

function App() {
  const [choices, setChoices] = useState({
    yearGroup: "",
    subject: "",
    topic: "",
    curriculumAim: "",
  });

  const [topics, setTopics] = useState({});
  const [curriculumAims, setCurriculumAims] = useState({});
  const [plan, setPlan] = useState({});

  const router = createBrowserRouter([
    {
      path: "/",
      element: <Root />,
      children: [
        {
          path: "/",
          element: <LandingPage />,
        },
        {
          path: "year-group",
          element: <YearGroup choices={choices} setChoices={setChoices} />,
        },
        {
          path: "subject",
          element: (
            <Subject
              choices={choices}
              setChoices={setChoices}
              topics={topics}
              setTopics={setTopics}
            />
          ),
        },
        {
          path: "topic",
          element: (
            <Topic
              choices={choices}
              setChoices={setChoices}
              topics={topics}
              curriculumAims={curriculumAims}
              setCurriculumAims={setCurriculumAims}
            />
          ),
        },
        {
          path: "curriculum-aim",
          element: <CurriculumAim curriculumAims={curriculumAims} />,
        },
        {
          path: "lesson-plan",
          element: (
            <LessonPlan choices={choices} plan={plan} setPlan={setPlan} />
          ),
        },
      ],
    },
  ]);

  return <RouterProvider router={router} />;
}

export default App;
