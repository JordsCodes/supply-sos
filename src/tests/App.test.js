import React from "react";
import App from "../components/App";
import { render, screen } from "@testing-library/react";

describe("App", () => {
  test("renders correctly", () => {
    const { asFragment } = render(<App />);

    expect(asFragment()).toMatchSnapshot();
  });
});
