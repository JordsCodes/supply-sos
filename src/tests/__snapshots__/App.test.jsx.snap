// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <h1
    class="heading"
  >
    SupplySoS
  </h1>
  <div
    class="landing-page"
  >
    <div
      class="welcome"
    >
      <h2
        class="welcome_heading"
      >
        No planning? No problem.
      </h2>
      <p
        class="welcome_text"
      >
        Downloadable, customisable AI-generated lesson plans for Key Stages 1 and 2.
      </p>
      <a
        class="button"
        href="/year-group"
      >
        Let's get started
      </a>
    </div>
    <img
      alt="teacher in classroom"
      class="image"
      src="landing-page.png"
    />
  </div>
</DocumentFragment>
`;
